### PR TITLE
Permit password as Sensitive string

### PR DESCRIPTION
**Pull request by jcpunk**
I'm casting all my passwords as Sensitive within `hiera`.  This should permit a Sensitive password without breaking the existing user base.

### CODE REVIEW

File: manifests/user.pp [hunk 1]
```diff
@@ -18,7 +18,7 @@
 #   SuperMicro IPMI, user id 2 is reserved for the ADMIN user.
 #
 define ipmi::user (
-  String $password,
+  Variant[Sensitive[String[1]], String[1]] $password,
```
Comment (jhoblitt) [inline manifests/user.pp:21]: Please add a unit test that passes a `sensitive()` value.

**Review by jhoblitt — CHANGES_REQUESTED** [review]

File: manifests/user.pp (hunk 1 above)
Comment (jcpunk) [reply manifests/user.pp:21]: Test provided... I think.

**Comment by jcpunk** [comment]
I just went through my notes and noticed `unwrap` for a string requires a newish puppet.  Added backwards compat logic.

File: manifests/user.pp [hunk 2]
```diff
@@ -33,6 +33,13 @@
     default: { fail('invalid privilege level specified') }
   }
 
+  if $password =~ Sensitive {
```
Comment (jhoblitt) [inline manifests/user.pp:36]: :+1:

**Review by jhoblitt — APPROVED** [review]
LGTM

**Pull request merged**